main/test/ca/uwaterloo/flix/library/TestStringBuilder.flix

```diff
@@ -0,0 +1,346 @@
+@test
+def appendInt1601(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt16(sb, 0i16);
+    StringBuilder.toString(sb) `assertEq!` "0"
+
+@test
+def appendInt1602(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt16(sb, 1i16);
+    StringBuilder.toString(sb) `assertEq!` "1"
+
+@test
+def appendInt1603(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt16(sb, -1i16);
+    StringBuilder.toString(sb) `assertEq!` "-1"
+
+/////////////////////////////////////////////////////////////////////////////
+// appendInt32                                                             //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def appendInt3201(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt32(sb, 0);
+    StringBuilder.toString(sb) `assertEq!` "0"
+
+@test
+def appendInt3202(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt32(sb, 1);
+    StringBuilder.toString(sb) `assertEq!` "1"
+
+@test
+def appendInt3203(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt32(sb, -1);
+    StringBuilder.toString(sb) `assertEq!` "-1"
+
+/////////////////////////////////////////////////////////////////////////////
+// appendInt64                                                             //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def appendInt6401(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt64(sb, 0i64);
+    StringBuilder.toString(sb) `assertEq!` "0"
+
+@test
+def appendInt6402(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt64(sb, 1i64);
+    StringBuilder.toString(sb) `assertEq!` "1"
+
+@test
+def appendInt6403(): Bool = 
+    let sb = StringBuilder.new();
+    StringBuilder.appendInt64(sb, -1i64);
+    StringBuilder.toString(sb) `assertEq!` "-1"
+
+/////////////////////////////////////////////////////////////////////////////
+// appendBigInt                                                            //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def appendBigInt01(): Bool =
+    let sb = StringBuilder.new();
+    StringBuilder.appendBigInt(sb, 0ii);
+    StringBuilder.toString(sb) `assertEq!` "0"
+
+@test
+def appendBigInt02(): Bool =
+    let sb = StringBuilder.new();
+    StringBuilder.appendBigInt(sb, 1ii);
+    StringBuilder.toString(sb) `assertEq!` "1"
+
+@test
+def appendBigInt03(): Bool =
+    let sb = StringBuilder.new();
+    StringBuilder.appendBigInt(sb, -1ii);
+    StringBuilder.toString(sb) `assertEq!` "-1"
+
+/////////////////////////////////////////////////////////////////////////////
+// appendFloat32                                                           //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def appendFloat3201(): Bool = 

```

Comment:
What about NaN and Inf/-Inf?

(Probably these are also poorly supported in Flix already... so maybe a TODO will suffice. I guess you can get NaN from `0.0/0.0` and Inf in similar spirit.)